code/game/objects/structures/ladders.dm

```diff
@@ -197,3 +197,61 @@
 
 /obj/structure/ladder/crafted
 	crafted = TRUE
+
+/obj/structure/ladder/unbreakable/teleport
+	name = "mysterious ladder"
+	desc = "A mysterious ladder leading down. Where does it go?"
+
+/obj/structure/ladder/unbreakable/teleport/Initialize(mapload, obj/structure/ladder/up, obj/structure/ladder/down)
+	AddElement(/datum/element/undertile, TRAIT_T_RAY_VISIBLE, use_anchor = TRUE)
+	return ..()
+
+/obj/structure/ladder/unbreakable/teleport/LateInitialize()
+	if(down)
+		return
+
+	shuffle(GLOB.ladders)
+	for (var/obj/structure/ladder/unbreakable/teleport/ladder in GLOB.ladders)
+		if(ladder == src || ladder.down)
+			continue
+		down = ladder
+		ladder.down = src
+		ladder.update_appearance()
+		update_appearance()
+		break
+
+	if(!down)
+		qdel(src, force = TRUE)
+
+/obj/structure/ladder/unbreakable/teleport/travel(going_up, mob/user, is_ghost, obj/structure/ladder/ladder)
+	var/turf/target = get_turf(ladder)
+
+	if(!is_ghost)
+		ladder.add_fingerprint(user)
+
+		if(isclosedturf(target))
+			to_chat(user, span_warning("[ladder] is blocked!"))
+			return
+
+		to_chat(user, span_notice("You climb down [src]..."))
+
+		user.zMove(target = target, z_move_flags = ZMOVE_CHECK_PULLEDBY|ZMOVE_ALLOW_BUCKLED|ZMOVE_INCLUDE_PULLED)
+
+		if(target.overfloor_placed)
+			var/turf/open/floor/floorturf = target
```

Comment:
this is a pretty scary assumption that the destination's turf will be a floor. if it's not, this code will break when it tries spawning the floortile that nonfloors don't have.